### ISSUE TITLE
False positive in S1854 (Dead stores)

### ISSUE THREAD

**Issue by felipebz** [SonarSource-VisualStudio/sonarlint-roslyn-analyzer#405]
Sample code:

```csharp
void Test()
{
    string description = null;
    try
    {
        description = "step 1"; // Remove this useless assignment to local variable "description".
        foo();

        description = "step 2";
        bar();
    }
    catch (Exception e)
    {
        log($"Error on {description}: {e.Message}");
    }
}
```

The first assignment isn't useless in this context, because the variable "description" is used in the catch clause. Its value changes depending on which method throws the exception.

**Comment by tvsonar** [comment]
Thanks @felipebz, this is something that's not easy to fix. We probably have to live with it for a while.
